tests/component.js

```diff
@@ -0,0 +1,351 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import Vue from 'vue';
+import { mount } from '@vue/test-utils';
+import CKEditorComponent from '../src/ckeditor';
+import sinon from 'sinon';
+import { getEditorNamespace } from '../src/utils/geteditornamespace';
+
+/* global window */
+const CKEDITOR = window.CKEDITOR;
+
+describe( 'CKEditor Component', () => {
+	const spies = {};
+
+	let skipReady = false;
+	let sandbox, wrapper, component, props;
+
+	beforeEach( done => {
+		spies.replace = sinon.spy( CKEDITOR, 'replace' );
+		spies.inline = sinon.spy( CKEDITOR, 'inline' );
+
+		wrapper = createComponent( props );
+		component = wrapper.vm;
```

Comment:
TBH I'm not keen on doing such things outside of the test. IMO `beforeEach` should only create the environment to run the test. However the component being the subject of the test should be created directly inside the test.

The creation of the component can be exported to the function.

Reply:
I don't have a real issue with the code (note that it wasn't written by me) where SUT is created in setup function. It seems like a pretty popular practice (see CKEditor4 tests where an editor is also created in setup functions, we just don't see it in unit tests) and reduces the complexity of the unit tests. I agree it may be a bit more convenient to have SUT building function, but it's not worth refactoring the whole test file at this moment.